src/compiler/emitter.ts

```diff
@@ -1257,26 +1257,34 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
 
                     // Children
                     if (children) {
-                        for (let i = 0; i < children.length; i++) {
-                            // Don't emit empty expressions
-                            if (children[i].kind === SyntaxKind.JsxExpression && !((<JsxExpression>children[i]).expression)) {
-                                continue;
+                        const isChildEmittable = (child: JsxChild): boolean => {
+                            if (child.kind === SyntaxKind.JsxExpression) {
+                                // Don't emit empty expressions
+                                return !!(<JsxExpression>child).expression;
+
+                            }
+                            else if (child.kind === SyntaxKind.JsxText) {
+                                // Don't emit empty strings
+                                return !!getTextToEmit(<JsxText>child);
                             }
 
-                            // Don't emit empty strings
-                            if (children[i].kind === SyntaxKind.JsxText) {
-                                const text = getTextToEmit(<JsxText>children[i]);
-                                if (text !== undefined) {
-                                    write(", \"");
-                                    write(text);
-                                    write("\"");
-                                }
+                            return true;
+                        };
+
+                        // build list of valid emittable jsx children
+                        const emittableChildren = children.filter(isChildEmittable);

```

Comment:
do we need the extra array allocation? it seems to me that you can achieve the same output without doing this.